packages/ethers/src/client.ts

```diff
@@ -421,21 +421,26 @@ export class Web3Modal extends Web3ModalScaffold {
 
         throw new Error('Contract method is undefined')
       },
-
       getEnsAddress: async (value: string) => {
-        const { chainId } = EthersStoreUtil.state
-        if (chainId && chainId === 1) {
-          const ensProvider = new InfuraProvider('mainnet')
+        try {
+          const chainId = NetworkUtil.caipNetworkIdToNumber(this.getCaipNetwork()?.id)
+          let ensName: string | null = null
+          let wcName: boolean | string = false
 
-          const name = await ensProvider.resolveName(value)
-          if (name) {
-            return name
+          if (value?.endsWith('wcn.id')) {
```

Comment:
Same, seems like we are calling `wcn.id` in multiple places. Calling them like `CONSTANTS_UTIL.ENS_NAME` or something would be really nice